lib/buildkite/test_collector/uploader.rb

```diff
@@ -54,5 +64,22 @@ def self.configure
     def self.tracer
       Thread.current[:_buildkite_tracer]
     end
+
+    def self.upload(data)
+      return false unless Buildkite::TestCollector.api_token
+
+      http = Buildkite::TestCollector::HTTPClient.new(Buildkite::TestCollector.url)
+
+      Thread.new do
+        response = begin
+          upload_attempts ||= 1
+          http.post_json(data)
+        rescue *Buildkite::TestCollector::Uploader::RETRYABLE_UPLOAD_ERRORS => e
+          if (upload_attempts += 1) < MAX_UPLOAD_ATTEMPTS
+            retry
```

Comment:
I figured out how to test threaded code with this one weird trick and discovered this might be off by one. We could resolve this by setting upload_attempts to 0 on line 75